src/ApiCore/AuthWrapper.php

```diff
@@ -57,15 +60,49 @@ public function __construct(FetchAuthTokenInterface $fetchAuthTokenInterface, ca
      */
     public function getBearerString()
     {
-        return 'Bearer ' . $this->getToken();
+        return 'Bearer ' . self::getToken($this->fetchAuthTokenInterface, $this->authHttpHandler);
+    }
+
+    /**
+     * @return callable Callable function that returns an authorization header.
+     */
+    public function getAuthorizationHeaderCallback()
+    {
+        $fetchAuthTokenInterface = $this->fetchAuthTokenInterface;
+        $authHttpHandler = $this->authHttpHandler;
+
+        // NOTE: changes to this function should be treated carefully and tested thoroughly. It will
+        // be passed into the gRPC c extension, and changes have the potential to trigger very
+        // difficult-to-diagnose segmentation faults.
+        return function () use ($fetchAuthTokenInterface, $authHttpHandler) {
+            return ['authorization' => ['Bearer ' . self::getToken($fetchAuthTokenInterface, $authHttpHandler)]];
+        };
     }
 
-    private function getToken()
+    private static function getToken($fetchAuthTokenInterface, $authHttpHandler)
     {
-        $token = $this->fetchAuthTokenInterface->getLastReceivedToken();
-        if (is_null($token) || time() > $token['expires_at']) {
-            $token = $this->fetchAuthTokenInterface->fetchAuthToken($this->authHttpHandler);
+        $token = $fetchAuthTokenInterface->getLastReceivedToken();
+        if (self::isExpired($token)) {
+            $token = $fetchAuthTokenInterface->fetchAuthToken($authHttpHandler);
         }
-        return $token['access_token'];
+        if (self::isValid($token)) {
+            return $token['access_token'];
+        } else {
+            return '';
+        }
+    }
+
+    private static function isValid($token)
+    {
+        return !is_null($token)
```

Comment:
I think the `is_array` check will be enough, it should return false if a null value is passed in.